xsl/makeSearchPage.xsl

```diff
@@ -608,6 +610,37 @@
         </xsl:choose>
     </xsl:function>
     
+    <xd:doc>
+        <xd:desc><xd:ref name="hcmc:getFilterLabel" type="function">hcmc:getFilterLabel</xd:ref> retrieves
+        an item to be used as the label for a filter on the search page. This is either going to be the 
+        text string which is the filter name from the html:meta/@name attribute, or (if the user has 
+        overridden this in their configuration file) an HTML label element with perhaps some HTML markup
+        inside it.</xd:desc>
+        <xd:param name="filterName" as="xs:string">The string value of the @name attribute, which can 
+        be used to look up the result.</xd:param>
+        <xd:param name="filterId" as="xs:string">The id value of the filter as it will be rendered on the page.</xd:param>
+        <xd:return as="element()">An HTML label element.</xd:return>
+    </xd:doc>
+    <xsl:function name="hcmc:getFilterLabel" as="element()">
+        <xsl:param name="filterName" as="xs:string"/>
+        <xsl:param name="filterId" as="xs:string"/>    
+        <xsl:message expand-text="yes">Filter name: {$filterName}; filter id: {$filterId}; filters to check: {count($filterLabels)}</xsl:message>
+        <xsl:choose>
+            <xsl:when test="$filterLabels[@filterName=$filterName and (@lang=$pageLang or not(@lang))]">
+                <xsl:variable name="currLabel" select="$filterLabels[@filterName=$filterName and (@lang=$pageLang or not(@lang))][1]/*[1]"/>
```

Comment:
Any reason why we can't specify this directly? (And maybe move the variable up to use for the `<xsl:when>`? E.g.

```xsl
<xsl:variable name="currLabel" 
select="$filterLabels[@filterName = $filterName and (@lang = $pageLang or not(@lang))][1]/xh:label[1]"
as="element(xh:label)?"/>
<xsl:choose>
<xsl:when test="$currLabel">
<!-- ... -->
</xsl:when>
</xsl:choose>



```

Reply:
Sure, reworked to follow that model. Also I tested on Windows and the build worked and produced a working test site. Once it a while it's good to do that.